native/src/routes/jpal-tracking/components/JpalTracking.js

```diff
@@ -0,0 +1,119 @@
+import AppSettings, { defaultSettings } from '../../../modules/settings/AppSettings'
+import type { NavigationPropType, RoutePropType } from '../../../modules/app/constants/NavigationTypes'
+import type { JpalTrackingRouteType } from 'api-client'
+import styled from 'styled-components/native'
+import LayoutContainer from '../../../modules/layout/containers/LayoutContainer'
+
+const ThemedText = styled.Text`
+  display: flex;
+  text-align: left;
+  color: ${props => props.theme.colors.textColor};
+  font-family: ${props => props.theme.fonts.decorativeFontRegular};
+  padding: 10px 0;
+`
+
+const DescriptionContainer = styled.View`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  padding: 15px 0 5px;
+`
+
+const Input = styled(TextInput)`
+  padding: 15px;
+  border-width: 1px;
+  border-color: ${props => props.theme.colors.themeColor};
+  text-align-vertical: top;
+  height: 50px;
+`
+
+export type PropsType = {|
+  theme: ThemeType,
+  t: TFunction,
+  route: RoutePropType<JpalTrackingRouteType>,
+  navigation: NavigationPropType<JpalTrackingRouteType>
+|}
+
+const JpalTracking = (props: PropsType, state: StateType) => {
+  const [appSettings] = useState(new AppSettings())
+  const [settings, setSettings] = useState(defaultSettings)
+  const [settingsLoaded, setSettingsLoaded] = useState(false)
+  const [settingsUpdated, setSettingsUpdated] = useState(false)
+  const [displayedTrackingCode, setDisplayedTrackingCode] = useState('')
+  const routeTrackingCode = props.route.params.trackingCode
+
+  useEffect(() => {
+    if (routeTrackingCode) {
+      appSettings.setJpalTrackingCode(routeTrackingCode)
+    }
+  }, [appSettings, routeTrackingCode])
+
+  useEffect(() => {
+    const loadSettings = async () => {
+      try {
+        const loadedSettings = await appSettings.loadSettings()
+        setSettingsLoaded(true)
+        setSettingsUpdated(false)
+        setSettings(loadedSettings)
+        setDisplayedTrackingCode(loadedSettings.jpalTrackingCode)
+      } catch (e) {
+        console.error('Failed to load settings.')
+      }
+    }
+    loadSettings()
+  }, [appSettings, settingsUpdated])
+
+  const onPress = () => {
+    appSettings.setJpalTrackingEnabled(!settings.jpalTrackingEnabled).then(setSettingsUpdated(true))
+  }
+
+  const setTrackingCode = (value: string) => {
+    setDisplayedTrackingCode(value)
+    appSettings.setJpalTrackingCode(value).then(setSettingsUpdated(true))
+    props.navigation.setParams({ trackingCode: value })
+  }
+
+  if (!settingsLoaded) {
+    return <LayoutContainer />
+  }
+
+  const { t, theme } = props
+  const { jpalTrackingEnabled } = settings
+
+  return (
+    <View style={{ padding: 40 }}>
+      <Caption title={t('tracking')} theme={theme} />
+      <Text>{t('trackingDescription')}</Text>
+
+      <DescriptionContainer>
+        <ThemedText theme={props.theme}>{t('trackingAllowed')}</ThemedText>
+        <Switch
```

Comment:
Maybe it makes sense to switch on a click on the text as well? I.e. wrap the text in the switch.

Reply:
So clicking on the text field would be a consent for tracking? Not sure if this is a good idea.

Reply:
I think that is the default behavior in apps, clicking somewhere in the row always toggles settings. Maybe we need a dialog at a later time depending on what texts, terms and conditions jpal gives us.

Reply:
I think we talk about different things. 
But now i get what you mean: The whole row and not only the switch should be clickable?